init.lua

```diff
@@ -811,18 +837,21 @@ local function update_form(self, player, form_info)
 end
 
 function Form:update(player)
-    local form_info = open_formspecs[player:get_player_name()]
+    local player_name = player:get_player_name()
+    local form_info = open_formspecs[player_name] or open_inv_formspecs[player_name]
     if form_info and form_info.self == self then
         update_form(self, player, form_info)
     end
 end
 
 function Form:update_where(func)
-    for name, form_info in pairs(open_formspecs) do
-        if form_info.self == self then
-            local player = minetest.get_player_by_name(name)
-            if player and func(player, form_info.ctx) then
-                update_form(self, player, form_info)
+    for _ , open_formspec_table in ipairs({open_formspecs, open_inv_formspecs}) do
+        for name, form_info in pairs(open_formspec_table) do
+            if form_info.self == self then
+                local player = minetest.get_player_by_name(name)
+                if player and func(player, form_info.ctx) then
+                    update_form(self, player, form_info)
+                end
```

Comment:
Good point. There's actually a bug I missed here where it will call `minetest.show_formspec` on the updated form. Not sure if that's how it's supposed to be done.

Reply:
fixed